### ISSUE TITLE
Cypress crashes after selecting project when back button is pressed

### ISSUE THREAD

**Issue by panzarino** [cypress-io/cypress#15089]
### Current behavior
When selecting a project in open mode, the application will crash if the back button is selected before the project has loaded. See gif below for an example:

![ezgif-2-121902ee6be6](https://user-images.githubusercontent.com/7033952/107910093-a5bd8200-6f27-11eb-88a6-9d4bf1a653a4.gif)

Error stack:

```
Error: ServerE2E#open must first be called before accessing 'this.server'
    at ServerE2E.ensureProp (/Users/panzarino/Documents/GitHub/cypress/packages/server/lib/util/class-helpers.ts:13:15)
    at ServerE2E.get server [as server] (/Users/panzarino/Documents/GitHub/cypress/packages/server/lib/server-base.ts:84:21)
    at ServerE2E._port (/Users/panzarino/Documents/GitHub/cypress/packages/server/lib/server-base.ts:179:21)
    at ServerE2E._onDomainSet (/Users/panzarino/Documents/GitHub/cypress/packages/server/lib/server-base.ts:259:72)
    at ServerE2E.reset (/Users/panzarino/Documents/GitHub/cypress/packages/server/lib/server-base.ts:325:21)
    at ServerE2E._close (/Users/panzarino/Documents/GitHub/cypress/packages/server/lib/server-base.ts:328:14)
    at ServerE2E.close (/Users/panzarino/Documents/GitHub/cypress/packages/server/lib/server-base.ts:343:45)
    at ProjectE2E.close (/Users/panzarino/Documents/GitHub/cypress/packages/server/lib/project-base.ts:158:99)
    at ProjectE2E.close (/Users/panzarino/Documents/GitHub/cypress/packages/server/lib/project-e2e.ts:65:22)
    at Object.closeOpenProjectAndBrowsers (/Users/panzarino/Documents/GitHub/cypress/packages/server/lib/open_project.js:279:35)
    at Object.close (/Users/panzarino/Documents/GitHub/cypress/packages/server/lib/open_project.js:293:19)
    at IpcMainImpl.handleEvent (/Users/panzarino/Documents/GitHub/cypress/packages/server/lib/gui/events.js:316:26)
    at apply (/Users/panzarino/Documents/GitHub/cypress/node_modules/lodash/lodash.js:478:17)
    at IpcMainImpl.wrapper (/Users/panzarino/Documents/GitHub/cypress/node_modules/lodash/lodash.js:5337:16)
    at IpcMainImpl.emit (events.js:315:20)
    at Object.<anonymous> (electron/js2c/browser_init.js:157:9051)
    at Object.emit (events.js:315:20)
```

### Versions

I've done some testing and discovered that this bug was introduced in https://github.com/cypress-io/cypress/commit/af26fbebe6bc609132013a0493a116cc78bb1bd4. It's not yet been released at the time of this issue being filed but will most likely appear for the first time in the next version (`6.5.0` or `7.0.0`)

**Comment by flotwig** [comment]
Same issue over here when reloading the project via desktop-gui after a plugin error:

![image](https://user-images.githubusercontent.com/1151760/112359425-27dc5b80-8cc9-11eb-8657-55e6d5460b6e.png)

**Comment by lmiller1990** [comment]
We could fix it like this: https://github.com/cypress-io/cypress/pull/15651

**Comment by bahmutov** [comment]
released in 7.0.0
